### ISSUE TITLE
Lights still come on if sleep state true but no sleep lights defined

### ISSUE THREAD

**Issue by sjthespian** [jseidl/hass-magic_areas#127]
I have two areas where I do not want the lights to come on when the presence is true after I set my house in "night mode" (i.e. we have gone to bed). However, I have found that If I set an areas to have a sleep state entity but do not set any sleep lights, it still turns on the main lights. For example with these settings:
<img width="584" alt="Screen Shot 2021-07-04 at 22 11 42" src="https://user-images.githubusercontent.com/1789285/124420734-d9e9e180-dd14-11eb-9690-d488e70e7f4a.png">
and with sleep state true:
<img width="1253" alt="Screen Shot 2021-07-04 at 21 24 31" src="https://user-images.githubusercontent.com/1789285/124420790-f2f29280-dd14-11eb-8064-0691aa08c044.png">
The main hall lights still come on when the hall motion sensor triggers.

I may be able to workaround this for one of my areas, as I have a nightlight on a switch module that I can create a light device for with a template (it looks like Magic Areas doesn't let me add a switch device). But this is something of a kludge and I don't have a good substitute for the other area(s).

**Comment by jseidl** [comment]
Hmmm interesting. I didn't developed that use case (sleep mode with no lights). Sleep mode does expect lights to be defined otherwise it's ignored by automatic lights and it just treats as if the sleep state wasn't there.

The workaround is, as you did, create a fake light with an input boolean and a template light and set that to the sleep light.

I'm currently migrating this whole logic to a new one which this **might** be possible. Will add to my test cases

**Comment by sjthespian** [comment]
I just created a couple of fake "sleep" lights, I should know in a few hours if this is a usable workaround. I'm hoping this will also fix something else I noticed -- I have lights that aren't in my "main lights" list getting turned off in the same area, and I think it is only happening in areas with an empty list of sleep lights.

Here's an example of a room where I have night mode disabled (taken from core.config_entries since I'm using the integration UI):
```
                "options": {
                    "features": [
                        "control_lights"
                    ],
                    "include_entities": [
                        "binary_sensor.great_room_motion",
                        "sensor.great_room_mot"
                    ],
                    "exclude_entities": [],
                    "presence_sensor_device_class": [
                        "motion",
                        "occupancy",
                        "presence"
                    ],
                    "clear_timeout": 300,
                    "icon": "mdi:texture-box",
                    "aggregates_min_entities": 2,
                    "notification_devices": [],
                    "notify_on_sleep": false,
                    "night_entity": "binary_sensor.dark_outside_projector_off",
                    "night_state": "on",
                    "main_lights": [
                        "light.great_room_lamp_bulb"
                    ],
                    "sleep_lights": [
                        "light.great_room_sleep"
                    ],
                    "sleep_entity": "binary_sensor.night_mode",
                    "sleep_state": "on",
                    "sleep_timeout": 0,
                    "update_interval": 60,
                    "type": "interior"
                },
```

Even with that config, I have other lights in the area that are being turned off, and not even at night. For example light.greatroom_esp_rgb_led was just turned off when the room presence went false even though both night and sleep are currently false. I could add them to the exclude list, but my expectation was that Magic Areas would only touch lights that were in the `main_lights` list.

**Comment by jseidl** [comment]
Yeah that's not how it works. Magic Areas will turn of _ALL_ lights in the area when it goes clear. `main_lights` is only used to turn it on. Currently if you want MagicAreas to not touch something, you have to exclude it.

I've confirmed that #126 will indeed do what you want of not turning any lights on if sleep is defined and it's on sleep state. Coming up on 3.0.0!

**Comment by sjthespian** [comment]
I'm going through the list of lights defined in each area and adding the ones that I don't want Magic Areas touching to the exclude list. I think that should take care of the problem with extra lights going off. The only thing that is going to be annoying is that I have to remember to do that any time I add a new light.

**Comment by sjthespian** [comment]
Just in case this comes up for anyone else, here is how I defined my "fake" sleep light(s):

``` yaml
light:
- platform: template
  lights:
    hall_sleep:
      friendly_name: Hall Sleep
      unique_id: template_hall_sleep
      turn_on:
        service: input_boolean.turn_on
        entity_id: input_boolean.hall_sleep
      turn_off:
        service: input_boolean.turn_off
        entity_id: input_boolean.hall_sleep
      set_level:
        service: input_number.set_value
        data_template:
          value: "{{ brightness }}"
          entity_id: input_number.hall_sleep
input_number:
  hall_sleep:
    min: 0
    max: 255
input_boolean:
  hall_sleep:
```

**Comment by jseidl** [comment]
BTW if you really want this to be kinda `null` action, you can omit the `input_number` and `set_level` as it's not required.

Also you might be able to get away with not additional entities with a bogus service call:

```
light:
- platform: template
  lights:
    hall_sleep:
      friendly_name: Hall Sleep
      unique_id: template_hall_sleep
      value_template: "off"
      turn_on:
        service: homeassistant.update_entity
        entity_id: light.hall_sleep
      turn_off:
        service: homeassistant.update_entity
        entity_id: light.hall_sleep
```

You can use other service calls such as `mqtt.publish`.

**Comment by jseidl** [comment]
I believe this is fixed on `3.0.0`, can you check?
